src/components/MapController/DualMapController/DualMapController.js

```diff
@@ -123,123 +131,77 @@ export default class MapController extends React.Component {
-
-    // generate the list of available times for one or two variable+run+period combinations.
-    // which may include multiple files of different time resolutions
-    // (annual, seasonal, or monthly).
-    let datasets = _.filter(props.meta,
-      { ensemble_member, start_date, end_date });
-
-    // if there is a comparison variable, get times from its datasets too
-    if (this.hasValidData('comparand', props)) {
-      datasets = datasets.concat(_.filter(props.comparandMeta,
-        { ensemble_member, start_date, end_date }));
-    }
-
-    const timesPromises =
-      datasets.map(ds => getTimeMetadata(ds.unique_id));
-
-    Promise.all(timesPromises).then(responses => {
-      let variableTimes = {};
-      let comparandTimes = {};
-
-      for (let i = 0; i < responses.length; i++) {
-        let id = Object.keys(responses[i].data)[0];
-        for (let timeidx in responses[i].data[id].times) {
-          var idxString = JSON.stringify({
-            timescale: responses[i].data[id].timescale,
-            timeidx,
-          });
-
-          // This assumes only one variable per file.
-          const variable = Object.keys(responses[i].data[id].variables)[0];
-          if (variable === props.meta[0].variable_id) {
-            variableTimes[idxString] = responses[i].data[id].times[timeidx];
-          }
-          if (this.hasValidData('comparand', props) &&
-            variable === props.comparandMeta[0].variable_id
-          ) {
-            comparandTimes[idxString] = responses[i].data[id].times[timeidx];
-          }
-        }
+    // and timestamp to the viewer component DataMap.
+    // The variable and the comparand may have different available timestamps, but will
+    // (aside from variable_id) display the same instance.
+
+    const { start_date, end_date, ensemble_member } = instance;
+    
+    let rasterParamsPromise = getRasterParamsPromise(instance, props.meta);
+    let isolineParamsPromise = getIsolineParamsPromise(instance, props.comparandMeta);
+    
+    Promise.all([rasterParamsPromise, isolineParamsPromise]).then(params => {
+            
+      let rasterParams = _.findWhere(params, {variableId: selectedVariable(props.meta)});
+      let isolineParams = _.findWhere(params, {variableId: selectedVariable(props.comparandMeta)});
+      
+      if(rasterParams === isolineParams) {
+        // needed when comparand and variable are the same
+        isolineParams = _.clone(rasterParams);
+      }
+      
+      // if the variable or comparand has changed, use the default values, otherwise
+      // the ones previously set by the user
+      if(!newVariable) {
+        rasterParams.palette = this.state.raster.palette;
+        rasterParams.logscale = this.state.raster.logscale;
+      }
+      if(!newComparand) {
+        isolineParams.palette = this.state.isoline.palette;
+        isolineParams.logscale = this.state.isoline.logscale;
+      }
+      
+      if(isolineParams.palette == 'x-Occam' && 
```

Comment:
Use `===`; avoid `==` ([it's type-unsafe](https://www.codementor.io/thinker3197/avoid-common-pitfalls-in-javascript-u0hzz2nfw)).

If you make one change to your JS habits, it should be this. `==` is Satan's tool.